cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/CbDelegatedTokenCredentials.java

```diff
@@ -122,14 +122,14 @@ public String clientId() {
      */
     public String generateAuthenticationUrl(String state) {
         return String.format("%s%s/oauth2/authorize?client_id=%s&response_type=code&redirect_uri=%s&response_mode=query&state=%s&resource=%s",
-                environment().activeDirectoryEndpoint(), domain(), clientId(), this.redirectUrl, state, environment().managementEndpoint());
+                environment().activeDirectoryEndpoint(), domain(), clientId(), redirectUrl, state, environment().managementEndpoint());
     }
 
     /**
      * Set the authorization code acquired returned to the redirect URL.
      * @param authorizationCode the oauth2 authorization code
      */
-    public void setAuthorizationCode(String authorizationCode) {
+    public synchronized void setAuthorizationCode(String authorizationCode) {
```

Comment:
why we need sync block for this?

Reply:
Because we used one of the variables inside in both synchronized and non synchronized context